dev_scripts/release-container-image.py

```diff
@@ -0,0 +1,246 @@
+IMAGE_NAME_PATH = PROJECT_ROOT / "share" / "image-name.txt"
+DOCKER_ENV_PATH = PROJECT_ROOT / "Dockerfile.env"
+
+
+def extract_debian_archive_date():
+    with DOCKER_ENV_PATH.open() as f:
+        content = f.read()
+
+    match = re.search(r"DEBIAN_ARCHIVE_DATE=(\d+)", content)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+
+def run(cmd, check=True, capture_output=False):
+    """Run a command and optionally capture output."""
+    click.echo(f"Running: {' '.join(cmd)}")
+    result = subprocess.run(cmd, check=check, capture_output=capture_output, text=True)
+    return result
+
+
+@click.command()
+@click.option("--commit", default=None)
+@click.option(
+    "--ghcr-signer-path",
+    type=click.Path(exists=True),
+    required=True,
+    help="Path to the ghcr-signer repository",
+    default="../ghcr-signer",
+)
+@click.option("--repository", default="freedomofpress/dangerzone")
+@click.option("--branch", default="main")
+def attest_release_sign(commit, ghcr_signer_path, repository, branch):
+    """
+    Attest the provenance of a commit, reproduce its digest and prepare signatures
+
+    Usage: release-container-image
+    """
+
+    # Get commit hash
+    if not commit:
+        result = run(
+            ["git", "rev-parse", "HEAD"],
+            capture_output=True,
+        )
+        commit = result.stdout.strip()
+
+    full_image = get_candidate_image(commit)
+    attest_provenance(full_image, commit, repository, branch)
+    digests = get_platform_digests(full_image)
+    debian_archive_date = extract_debian_archive_date()
+    for platform in ["linux/amd64", "linux/arm64"]:
+        pass  # reproduce_image(full_image, platform, debian_archive_date, digests[platform])
+
+    sign_image(full_image, ghcr_signer_path)
+
+
+def attest_provenance(full_image, commit, repository, branch):
+    click.echo(f"\n🔐 Attesting provenance for image: {full_image}")
+    click.echo(f"   Commit: {commit}\n")
+
+    try:
+        run(
+            [
+                "poetry",
+                "run",
+                "python",
+                "dev_scripts/dangerzone-image",
+                "attest-provenance",
+                "--repository",
+                repository,
+                "--branch",
+                branch,
+                "--commit",
+                commit,
+                full_image,
+            ]
+        )
+        click.echo("\n✅ Provenance attestation successful\n")
+    except subprocess.CalledProcessError as e:
+        click.echo(f"\n❌ Provenance attestation failed: {e}\n")
+        sys.exit(1)
+
+
+def get_candidate_image(commit):
+    short_commit = commit[:7]
+
+    image_name = IMAGE_NAME_PATH.read_text().strip()
+
+    click.echo(f"\n📦 Looking for images for commit: {short_commit}")
+    click.echo(f"   Repository: {image_name}\n")
+
+    # Get the latest container image with this commit
+    try:
+        result = run(
+            [
+                "crane",
```

Comment:
I think we need a preflight check at the very start, where we will ensure that `uv`, `crane`, `git` and `poetry` are in the `PATH`.

Reply:
Just did in 48757192388ad48578d40194ba66f38ebe5de715